src/main/clojure/clara/benchmark/accum_repeatedly_not_selected.clj

```diff
@@ -0,0 +1,24 @@
+(ns clara.benchmark.accum-repeatedly-not-selected
+  "Added to test performance improvements in clara-rules issue 182"
+  (:require [clara.rules :refer :all]
+            [clara.rules.accumulators :as acc])
+  (:import [clara.benchmark.beans Discount Customer]
+           [java.util
+            ArrayList
+            List]))
+
+(def min-id (acc/min (fn [^Customer c] (.getId c)) :returns-fact true))
+
+(defrule vip-discount
+  [?customer <- min-id :from [Customer]]
+  =>
+  (apply insert! (let [fact-list ^List (ArrayList.)]
+                   (loop [n 0]
+                     (.add fact-list (Discount. 10))
+                     (if (< n 1000)
+                       (recur (inc n))
+                       fact-list)))))
```

Comment:
The reason for using an ArrayList here isn't clear to me. Would something like the following work?

```(for [n (range 1000)] (Discount. 10))```